enhancements/update/change-management-and-maintenance-schedules.md

```diff
@@ -0,0 +1,1334 @@
+update will drain user workloads and will cause workload disruption.
+
+Though disruption is inevitable, the maintenance schedule feature provides value in these toplogies by
+giving the cluster lifecycle administrator high level controls on exactly when it occurs.
+
+#### OCM Managed Profiles
+OpenShift Cluster Manager (OCM) should expose a user interface allowing users to manage their change management policy.
+Standard Fleet clusters will expose the option to configure the control-plane and worker-node `ChangeManagementPolicy`
+objects with the `MaintenanceSchedule` strategy - including permit and exclude times.
+
+- Service Delivery will reserve the right to disable this policy for emergency corrective actions.
+- Service Delivery should constrain permit & exclude configurations based on their internal policies. 
+  For example, customers may be forced to enable permissive windows which amount to at least 6 hours a month.
+
+### Implementation Details/Notes/Constraints
+
+#### ChangeManagement Stanza
+The `spec.changeManagement` stanza will be introduced into ClusterVersion and MachineConfigPool (for standalone profiles)
+and HostedCluster and NodePool (for HCP profiles).
+
+#### MaintenanceSchedule Strategy Configuration
+
+When a `ChangeManagementPolicy` is defined to use the `MaintenanceSchedule` strategy,
+a `maintenanceSchedule` stanza should also be provided to configure the strategy
+(if it is not, the policy is functionally identical to `Restrictive`).
+
+```yaml
+kind: ChanageManagementPolicy
+spec:
+  strategy: MaintenanceSchedule
+ 
+  maintenanceSchedule:
+    # Specifies a reoccurring permissive window. 
+    permit:  
+      # RRULEs (https://www.rfc-editor.org/rfc/rfc5545#section-3.3.10) are commonly used 
+      # for calendar management metadata. Only a subset of the RFC is supported.
+      # See "RRULE Constraints" section for details.
+      # If unset, all dates are permitted and only exclude constrains permissive windows.
+      recurrence: <rrule|null>
+      # Given the identification of a date by an RRULE, at what time (always UTC) can the 
+      # permissive window begin. "00:00" if unset.
+      startTime: <time-of-day|null>
+      # Given the identification of a date by an RRULE, after what offset from the startTime should
+      # the permissive window close. This can create permissive windows within days that are not
+      # identified in the RRULE. For example, recurrence="FREQ=Weekly;BYDAY=Sa;",
+      # startTime="20:00", duration="8h" would permit material change initiation starting
+      # each Saturday at 8pm and continuing through Sunday 4am (all times are UTC). The default
+      # duration is 24:00-startTime (i.e. to the end of the day).
+      duration: <duration|null>
+      
+    
+    # Excluded date ranges override RRULE selections.
+    exclude:
+    # Dates should be specified in YYYY-MM-DD. Each date is excluded from 00:00 UTC for 24 hours.

```

Comment:
Is there any reason you'd want to exclude a window short than 24 hours, how could we model that?

Reply:
If it came in as an important RFE, the solution could be to change the time formats supported by `fromDate` and `toDate` to include HH:MM:SS type information. I couldn't think of a compelling use case to warrant the increased complexity. "Black Friday", "Someone is on vacation" -- lend themselves to day granularity. The day granularity is also a nod to the fact that I wanted this to be hard to get wrong. If you are going to the trouble to specify an exclude, making a mistake is probably a really bad outcome.